.drone.yml

```diff
@@ -3,31 +3,95 @@ kind: pipeline
 name: pipeline1
 
 steps:
+- name: restore-cache
+  image: plugins/s3-cache
+  failure: ignore
+  settings:
+    pull: true
+    access_key:
+      from_secret: drone_aws_access_key_id
+    secret_key:
+      from_secret: drone_aws_secret_access_key
+    restore: true
+    root: TODO-BUCKET-NAME
+    filename: pipeline1.tar
+
 - name: checkstyle-and-sevntu
   image: maven:3.6.3-adoptopenjdk-8
+  environment:
+    MAVEN_OPTS: "-Dmaven.repo.local=.m2"
   commands:
   - ./.ci/travis/travis.sh checkstyle-and-sevntu
 
 - name: spotbugs-and-pmd
   image: maven:3.6.3-adoptopenjdk-8
+  environment:
+    MAVEN_OPTS: "-Dmaven.repo.local=.m2"
   commands:
   - ./.ci/travis/travis.sh spotbugs-and-pmd
 
+- name: rebuild-cache
+  image: plugins/s3-cache
+  failure: ignore
+  settings:
+    pull: true
+    access_key:
+      from_secret: drone_aws_access_key_id
+    secret_key:
+      from_secret: drone_aws_secret_access_key
+    rebuild: true
+    root: TODO-BUCKET-NAME
+    filename: pipeline1.tar
+    mount:
+    - .m2
+
 ---
 kind: pipeline
 name: pipeline2
 
 steps:
+- name: restore-cache
+  image: plugins/s3-cache
+  failure: ignore
+  settings:
+    pull: true
+    access_key:
+      from_secret: drone_aws_access_key_id
+    secret_key:
+      from_secret: drone_aws_secret_access_key
+    restore: true
+    root: TODO-BUCKET-NAME
+    filename: pipeline2.tar
+
 - name: site
   image: maven:3.6.3-adoptopenjdk-8
+  environment:
+    MAVEN_OPTS: "-Dmaven.repo.local=.m2"
   commands:
   - ./.ci/travis/travis.sh site
 
 - name: release-dry-run
   image: maven:3.6.3-adoptopenjdk-8
+  environment:
+    MAVEN_OPTS: "-Dmaven.repo.local=.m2"
   commands:
   - ./.ci/travis/travis.sh release-dry-run
 
+- name: rebuild-cache
+  image: plugins/s3-cache
+  failure: ignore
+  settings:
+    pull: true
+    access_key:
+      from_secret: drone_aws_access_key_id
+    secret_key:
+      from_secret: drone_aws_secret_access_key
+    rebuild: true
```

Comment:
Do you know if this plugin skip sync of file file it it present (same) on s3 already ?
To avoid traffic if not required.

For us most preferable is to avoid upload if same by name is already present in s3

Reply:
Its quite simplistic in this aspect: creates a single archive and uploads it to S3, similar to:
```bash
tar -czvf pipeline1.tar .m2
aws s3 cp pipeline1.tar s3://bucket/
```
Which allows single request to download or upload, but no content per file comparison.

Reply:
I think minimal permissions AWS inline policy would look like:
```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": "s3:ListBucket",
            "Resource": "arn:aws:s3:::TODO-BUCKET-NAME"
        },
        {
            "Sid": "VisualEditor1",
            "Effect": "Allow",
            "Action": [
                "s3:PutObject",
                "s3:GetObject",
                "s3:GetBucketLocation",
                "s3:ListMultipartUploadParts"
            ],
            "Resource": [
                "arn:aws:s3:::TODO-BUCKET-NAME/*",
                "arn:aws:s3:::TODO-BUCKET-NAME"
            ]
        }
    ]
}
```

Reply:
@sukolenvo it is not about permissions. Github actions job upload files to s3 and it is fine. 
I believe @romani is asking in cases where there is no changes in cached dependencies, in this case we do not need to upload them to s3. So, if there is some check to know, is there cache or not to decide, should we upload it or not.

Reply:
we are using public read-only archive of artifacts for now, to avoid aws billing.
download of new version of libraries unlikely to be a big deal.